<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1021</width>
    <height>623</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Vision Par Ordinateur</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <property name="iconSize">
   <size>
    <width>24</width>
    <height>24</height>
   </size>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">QWidget{
	background-color: &quot;#fff&quot;
}</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="3" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <property name="rightMargin">
       <number>10</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QFrame" name="frame">
          <property name="styleSheet">
           <string notr="true">QFrame{
	background-color:&quot;#F1F1F1&quot;
}</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QPushButton" name="histButton">
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px
}</string>
             </property>
             <property name="text">
              <string>Hostogramme</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="negatifButton">
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px
}</string>
             </property>
             <property name="text">
              <string>Negatif</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="otsuButton">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>0</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
             </property>
             <property name="text">
              <string>OTSU</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="croppButton">
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
             </property>
             <property name="text">
              <string>Selectionner une partie</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="horizontalFrame">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>100</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">QFrame{
	background-color:&quot;#F1F1F1&quot;
}</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <property name="spacing">
               <number>12</number>
              </property>
              <property name="sizeConstraint">
               <enum>QLayout::SetFixedSize</enum>
              </property>
              <property name="topMargin">
               <number>1</number>
              </property>
              <item>
               <widget class="QLineEdit" name="cropXInput">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>80</width>
                  <height>100</height>
                 </size>
                </property>
                <property name="placeholderText">
                 <string>cropX</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLineEdit" name="cropYInput">
                <property name="maximumSize">
                 <size>
                  <width>80</width>
                  <height>100</height>
                 </size>
                </property>
                <property name="placeholderText">
                 <string>cropY</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="gaussFilter">
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
             </property>
             <property name="text">
              <string>Filtre Gaussien</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="gaussienInput">
             <property name="placeholderText">
              <string>Ã©cart type</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="medianFiltre">
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
             </property>
             <property name="text">
              <string>Filtre Median</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="medianInput">
             <property name="placeholderText">
              <string>taille de filtre</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="filtreMoyen">
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
             </property>
             <property name="text">
              <string>Filtre Moyenne</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="moyenneInput">
             <property name="placeholderText">
              <string>taille de filtre</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="egalisationButton">
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
             </property>
             <property name="text">
              <string>Egalisation</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="etirementButton">
             <property name="styleSheet">
              <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
             </property>
             <property name="text">
              <string>Etirement</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_6">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer_8">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="imageLabel">
            <property name="text">
             <string>Image</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_7">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="verticalFrame">
        <property name="styleSheet">
         <string notr="true">QFrame{
	background-color:&quot;#F1F1F1&quot;
}</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="leftMargin">
          <number>10</number>
         </property>
         <property name="topMargin">
          <number>10</number>
         </property>
         <property name="rightMargin">
          <number>10</number>
         </property>
         <property name="bottomMargin">
          <number>10</number>
         </property>
         <item>
          <widget class="QPushButton" name="gradientButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Gradient Contour</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="sobelButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Sobel Contour</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="prewittButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Prewitt Contour</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="laplacienButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Laplacien Contour</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="robertButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Robert  Contour</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="erodeButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Erosion</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="delateButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Delatation</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="ouvertureButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Ouverture</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="fermetureButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Fermeture</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="hitOrMissButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Hit or Miss</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="kmeansButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>k-means seg</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="nbrClusetInput">
           <property name="placeholderText">
            <string>nbr du cluster</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="croissRegButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Croissance de rÃ©gions</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="repartitionRegiButton">
           <property name="styleSheet">
            <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px;
}</string>
           </property>
           <property name="text">
            <string>Partision RÃ©gions</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="resetBtn">
      <property name="text">
       <string>RESET</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="open">
        <property name="text">
         <string>Open</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="save">
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="3" column="1" rowspan="2">
     <widget class="QWidget" name="widget" native="true">
      <property name="styleSheet">
       <string notr="true">QFrame{
	background-color: &quot;#F1F1F1&quot;
}</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Binarisation</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item alignment="Qt::AlignHCenter">
        <widget class="QSlider" name="binValueSlider">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Rotation</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item alignment="Qt::AlignHCenter">
        <widget class="QSlider" name="rotationSlider">
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="maximum">
          <number>360</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QFrame" name="horizontalFrame">
      <property name="styleSheet">
       <string notr="true">QFrame{
	background-color: &quot;#F1F1F1&quot;
}</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>60</height>
          </size>
         </property>
         <property name="text">
          <string>width</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="widthInput"/>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>60</height>
          </size>
         </property>
         <property name="text">
          <string>height</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="heightInput"/>
       </item>
       <item alignment="Qt::AlignRight">
        <widget class="QPushButton" name="resizeButton">
         <property name="styleSheet">
          <string notr="true">QPushButton{
	background-color:&quot;#555555&quot;;
	color:&quot;#fff&quot;;
	padding:6px;
	border-raduis: 5px
}</string>
         </property>
         <property name="text">
          <string>resize</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>save</sender>
   <signal>clicked()</signal>
   <receiver>imageLabel</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>26</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>open</sender>
   <signal>clicked()</signal>
   <receiver>imageLabel</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>83</x>
     <y>26</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetBtn</sender>
   <signal>clicked()</signal>
   <receiver>imageLabel</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>612</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
